### PR TITLE
[FINAL] Restore Inherit Color Functionality

### PR DESCRIPTION
**Pull request by jeiting**
This should satisfy everyone.

### CODE REVIEW

File: src/moaicore/MOAIColor.cpp [hunk 1]
```diff
@@ -227,6 +208,7 @@ void MOAIColor::RegisterLuaClass ( MOAILuaState& state ) {
 	state.SetField ( -1, "ATTR_OPACITY", MOAIColorAttr::Pack ( ATTR_OPACITY ));
 	
 	state.SetField ( -1, "ADD_COLOR", MOAIColorAttr::Pack ( ADD_COLOR ));
+	state.SetField ( -1, "INHERIT_COLOR_RAW", MOAIColorAttr::Pack ( INHERIT_COLOR_RAW ));
```
Comment (alexey-dc) [inline src/moaicore/MOAIColor.cpp:211]: Yeah, good call, I think this is a reasonable way to go.

Comment (jeiting) [inline src/moaicore/MOAIColor.cpp:211]: Aaron's call but it should meet everyones needs. Plus our better understanding allowed me to clean up the color code quite a bit.

**Comment by aab29** [comment]
I just glanced at this, but it all looks good.  :thumbsup: :ship:

**Comment by alexey-dc** [comment]
I like this, looks a lot cleaner... How are you doing the actual alpha premultiplication though? Can't find that...

**Comment by jeiting** [comment]
this->mColor.Modulate(USColorVec(this->mA,this->mA, this->mA, 1.0f));

**Comment by alexey-dc** [comment]
yeah, nice, I really like that, very elegant, I was hoping we'd end up using modulate for it :)

**Pull request merged**